core/views.py

```diff
@@ -47,6 +48,55 @@ def get_queryset(self: "SavedPostListView") -> models.QuerySet:
         )
 
 
+class UserPostListView(LoginRequiredMixin, ListView):
+    template_name = "core/user-post-list.html"
+    context_object_name = "posts"
+
+    def get_queryset(self: "UserPostListView") -> models.QuerySet:
+        user_post = Post.objects.filter(author=self.request.user).order_by("-created_at")
```

Comment:
We have a bug here. 
Everything is a Post object in the system.
A post with parent pointing to other post is comment. A post with parent = null is Post - start of a thread. So please add here condition to check parent = null. 
Look here: https://github.com/pymasterspl/reddit/blob/08e1ae4566f1f5c5cfb6aed5c09f0db3a0a41697/core/models.py#L38